src/config/theme/parser.rs

```diff
@@ -0,0 +1,891 @@
+use std::collections::HashMap;
+
+use chumsky::prelude::*;
+
+use super::{
+    StyleFile,
+    properties::{
+        PropertyFile,
+        PropertyKindFile,
+        PropertyKindFileOrText,
+        SongPropertyFile,
+        StatusPropertyFile,
+        WidgetPropertyFile,
+    },
+    style::Modifiers,
+};
+
+pub fn string_parser<'a>() -> impl Parser<'a, &'a str, String, extra::Err<Rich<'a, char>>> + Clone {
+    let escape_double_quoted_str = just('\\')
+        .then(choice((
+            just('\\'),
+            just('/'),
+            just('"'),
+            just('b').to('\x08'),
+            just('f').to('\x0C'),
+            just('n').to('\n'),
+            just('r').to('\r'),
+            just('t').to('\t'),
+            just('u').ignore_then(text::digits(16).exactly(4).to_slice().validate(
+                |digits, e, emitter| {
+                    char::from_u32(
+                        u32::from_str_radix(digits, 16)
+                            .expect("Only valid digits should have been parsed in text::digits"),
+                    )
+                    .unwrap_or_else(|| {
+                        emitter.emit(Rich::custom(e.span(), "invalid unicode character"));
+                        '\u{FFFD}'
+                    })
+                },
+            )),
+        )))
+        .ignored()
+        .boxed();
+
+    let escape_single_queoted_str =
+        choice((
+            just('\\').then(just('\'')).to('a').ignored(),
+            just('\\')
+                .then(choice((
+                    just('\\'),
+                    just('/'),
+                    just('b').to('\x08'),
+                    just('f').to('\x0C'),
+                    just('n').to('\n'),
+                    just('r').to('\r'),
+                    just('t').to('\t'),
+                    just('u').ignore_then(text::digits(16).exactly(4).to_slice().validate(
+                        |digits, e, emitter| {
+                            char::from_u32(u32::from_str_radix(digits, 16).expect(
+                                "Only valid digits should have been parsed in text::digits",
+                            ))
```

Comment:
Same thing, here i think its best to use chumsky for error reporting and exit gracefully